packages/flutter/test/widgets/tap_region_test.dart

```diff
@@ -1053,4 +1053,97 @@ void main() {
     await click(find.text('Outside Surface'));
     expect(tappedInside, isEmpty);
   });
+
+  testWidgets('TapRegion should register and unregister correctly on push and pop', (WidgetTester tester) async {
+    const ValueKey<String> tapRegion1Key = ValueKey<String>('TapRegion');
+    const ValueKey<String> tapRegion2Key = ValueKey<String>('TapRegion2');
+
+    int count1 = 0;
+    int count2 = 0;
+
+    final TapRegion tapRegion1 = TapRegion(
+      key: tapRegion1Key,
+      onTapOutside: (PointerEvent event) {
+        count1++;
+      },
+      child: Container(
+        width: 100,
+        height: 100,
+        color: Colors.red,
+      ),
```

Comment:
I probably shouldn't be suggesting this since it's not a big deal whatsoever, but I'm in the habit of avoiding Container objects since they have suboptimal performance.

```suggestion
      behavior: HitTestBehavior.opaque,
      child: SizedBox.square(dimension: 100),
```